src/NodaTime/Calendars/WondrousYearMonthDayCalculator.cs

```diff
@@ -0,0 +1,385 @@
+﻿// Copyright 2014 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using NodaTime.Calendars.Wondrous;
+using NodaTime.Utility;
+using System;
+
+namespace NodaTime.Calendars
+{
+    /// <summary>
+    ///     See <see cref="CalendarSystem.GetWondrousCalendar" /> for details.
+    /// </summary>
+    internal sealed class WondrousYearMonthDayCalculator : YearMonthDayCalculator
+    {
+        private const int WondrousMinYear = 1;
+        private const int WondrousMaxYear = 1000;
+
+        private const int AverageDaysPer10Years = 3652; // Ideally 365.2425 per year...
```

Comment:
Is this correct, or was it copied from somewhere?

Reply:
Likely copied.

Reply:
It is correct.